utils.go

```diff
@@ -35,3 +48,87 @@ func getFileContents(file string) (string, error) {
 
 	return string(bytes), nil
 }
+
+func getKernelVersion() (string, error) {
```

Comment:
FWIW, I'd just have used the uname syscall (man 2 uname) and https://golang.org/pkg/syscall/#Uname